packages/smooth_app/lib/widgets/tab_navigator.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/pages/history_page.dart';
+import 'package:smooth_app/pages/page_manager.dart';
+import 'package:smooth_app/pages/user_preferences_page.dart';
+
+class TabNavigator extends StatelessWidget {
+  const TabNavigator({required this.navigatorKey, required this.tabItem});
+  final GlobalKey<NavigatorState> navigatorKey;
+  final BottomNavigationTab tabItem;
+
+  @override
+  Widget build(BuildContext context) {
+    late Widget child;
```

Comment:
Here it's more `final` than `late`: we are 100% positive that the variable will be set.